src/blockchain/pop/vbk_block_tree.cpp

```diff
@@ -456,6 +452,43 @@ bool VbkBlockTree::isWeaklyEquivalent(const VTB& vtb1, const VTB& vtb2) {
           is_on_the_same_btc_chain);
 }
 
+int VbkBlockTree::weaklyCompare(const VTB& vtb1, const VTB& vtb2) {
+  VBK_ASSERT_MSG(areWeaklyEquivalent(vtb1, vtb2),
+                 "vtbs should be weakly equivalent");
+
+  auto* blockOfProof1 =
+      btc().getBlockIndex(vtb1.transaction.blockOfProof.getHash());
+  auto* blockOfProof2 =
+      btc().getBlockIndex(vtb2.transaction.blockOfProof.getHash());
+
+  VBK_ASSERT_MSG(blockOfProof1,
+                 "unknown block %s",
+                 vtb1.transaction.blockOfProof.toPrettyString());
+  VBK_ASSERT_MSG(blockOfProof2,
+                 "unknown block %s",
+                 vtb2.transaction.blockOfProof.toPrettyString());
+
+  int vtb1_score = 0;
+  int vtb2_score = 0;
+
+  vtb1_score += blockOfProof1->getHeight() > blockOfProof2->getHeight() ? 2 : 0;
+  vtb2_score += blockOfProof1->getHeight() < blockOfProof2->getHeight() ? 2 : 0;
+
+  vtb1_score += vtb1.transaction.publishedBlock.height >
+                        vtb2.transaction.publishedBlock.height
+                    ? 1
+                    : 0;
+  vtb2_score += vtb1.transaction.publishedBlock.height <
+                        vtb2.transaction.publishedBlock.height
+                    ? 1
+                    : 0;
+
+  VBK_ASSERT_MSG(
+      areStronglyEquivalent(vtb1, vtb2),
+      "if we can not define the best vtb they should be strongly equivalent");
```

Comment:
I do not see a condition before assert. This assert tells that all vtbs in this func must be strongly equal